lib/content-services/src/lib/search/components/search-date-range-advanced-tabbed/search-date-range-advanced-tabbed.component.ts

```diff
@@ -0,0 +1,121 @@
+import { SearchWidgetSettings } from '../../models/search-widget-settings.interface';
+import { SearchQueryBuilderService } from '../../services/search-query-builder.service';
+import { InLastDateType } from './search-date-range-advanced/in-last-date-type';
+
+@Component({
+  selector: 'adf-search-date-range-advanced-tabbed',
+  templateUrl: './search-date-range-advanced-tabbed.component.html'
+})
+export class SearchDateRangeAdvancedTabbedComponent implements SearchWidget {
+    displayValue$ = new Subject<string>();
+    id: string;
+    startValue: SearchDateRangeAdvanced = {
+        dateRangeType: DateRangeType.ANY,
+        inLastValueType: InLastDateType.DAYS,
+        inLastValue: undefined,
+        betweenStartDate: undefined,
+        betweenEndDate: undefined
+    };
+    settings?: SearchWidgetSettings;
+    context?: SearchQueryBuilderService;
+    fields: string[];
+    displayedLabelsByField: { [key: string]: string } = {};
+    queries: { [key: string]: string } = {};
+    valuesToDisplay: { [key: string]: string } = {};
+    tabsValidity: { [key: string]: boolean } = {};
+    combinedQuery: string;
+    combinedValuesToDisplay: string;
+
+    private value: { [key: string]: Partial<SearchDateRangeAdvanced> } = {};
+
+    getCurrentValue(): { [key: string]: Partial<SearchDateRangeAdvanced> } {
+        return this.value;
+    }
+
+    hasValidValue(): boolean {
+        return !Object.values(this.tabsValidity).some((valid) => !valid);
+    }
+
+    reset(): void {
+        this.queries = {};
+        this.valuesToDisplay = {};
+        this.startValue = {
+            ...this.startValue
+        };
+        setTimeout(() => {
+            this.submitValues();
+        });
+    }
+
+    setValue(value: { [key: string]: SearchDateRangeAdvanced }) {
+        this.value = value;
+    }
+
+    submitValues(): void {
+        this.context.queryFragments[this.id] = this.combinedQuery;
+        this.displayValue$.next(this.combinedValuesToDisplay);
+        if (this.id && this.context) {
+            this.context.update();
+        }
+    }
+
+    onDisplayValueUpdate(valueToDisplay: string, field: string) {
+        this.valuesToDisplay = {
+            ...this.valuesToDisplay,
+            [field]: valueToDisplay
+        };
+    }
+
+    onQueryUpdate(query: string, field: string) {
+        this.queries = {
+            ...this.queries,
+            [field]: query
+        };
+    }
+
+    onQueriesCombined(query: string) {
+        this.combinedQuery = query;
+    }
+
+    onValuesToDisplayCombined(valuesToDisplay: string) {
+        this.combinedValuesToDisplay = valuesToDisplay;
+    }
+
+    onTabValid(tabValid: boolean, field: string) {
+        this.tabsValidity[field] = tabValid;
+    }
+
+    onDateRangedValueChanged(value: Partial<SearchDateRangeAdvanced>, field: string) {
+        this.value[field] = value;
+    }
+
+    onFieldsChanged(fields: string[]) {
```

Comment:
Please move this assignment to html

Reply:
The code has been updated and the fields array is now calculated on init of the SearchDateRangeAdvancedTabbed component. This event no longer exists. So no need to add this code anymore